### ISSUE TITLE
unable to split book

### ISSUE THREAD

**Issue by Butterfly-Dragon** [openaudible/openaudible#765]
trying to split this book gives me an error and exits

the book
https://www.audible.com/pd/Immune-Audiobook/0593416317

the error
```
Selected 1 book(s) to split

• Error Immune Comparison method violates its general contract! 

 Good:0, Errors:1 Skipped:0
 Destination: [REDACTED]
```

**Comment by openaudible** [comment]
That’s a bug. Can you look in the console window (first file menu item) and look for a full error message regarding the “ Immune Comparison method violates its general contract!” Error? That is new and would like to fix.

**Comment by Butterfly-Dragon** [comment]
```
Nov. 04, 2021 5:14:44 P.M. org.openaudible.Directories getPrefs
INFO: Loading C:\Users\simon\OpenAudible\settings.json exists=true size=478
Nov. 04, 2021 5:14:44 P.M. org.openaudible.Directories getPrefs
INFO: loaded settings 1553606660536
Nov. 04, 2021 5:14:45 P.M. org.openaudible.Directories getPrefs
INFO: Loading C:\Users\simon\OpenAudible\settings.json exists=true size=478
Nov. 04, 2021 5:14:45 P.M. org.openaudible.Directories getPrefs
INFO: loaded settings 1553606660536
Nov. 04, 2021 5:14:45 P.M. org.openaudible.audible.AudibleConnectionState setRegion
INFO: Setting audible region to US
Nov. 04, 2021 5:14:45 P.M. org.openaudible.Audible load
INFO: Loading C:\Users\simon\OpenAudible\books.json
Nov. 04, 2021 5:14:45 P.M. org.openaudible.Audible checkLibrary
INFO: loaded 1844 books.
Nov. 04, 2021 5:14:46 P.M. org.openaudible.audible.AudibleConnectionState connectionChanged
INFO: Connection state changed to Connected from Not_Connected
Nov. 04, 2021 5:14:46 P.M. org.openaudible.audible.APIManager postInit
INFO: postInit API ready.
Nov. 04, 2021 5:14:46 P.M. org.openaudible.desktop.swt.manager.AudibleGUI applicationStarted
INFO: applicationStarted took 1728 ms. to complete.
Nov. 04, 2021 5:14:49 P.M. org.openaudible.desktop.swt.manager.VersionCheck getVersion
INFO: versionCheck: https://openaudible.org/swt_version.json
Nov. 04, 2021 5:15:00 P.M. org.openaudible.desktop.swt.manager.menu.CommandCenter execute
INFO: Command: Console
Nov. 04, 2021 5:16:09 P.M. org.openaudible.desktop.swt.manager.menu.CommandCenter execute
INFO: Command: Split_Selected_Books
Nov. 04, 2021 5:16:09 P.M. org.openaudible.util.SimpleProcess <init>
INFO: Running: "C:\Program Files\OpenAudible\bin\win\ffprobe.exe" -hide_banner C:\Users\simon\OpenAudible\books\Immune.mp3 -show_streams -show_chapters -show_error -show_format -print_format json 
Nov. 04, 2021 5:16:10 P.M. org.openaudible.convert.SplitAudio splitByChapter
SEVERE: failed to split
java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.base/java.util.TimSort.mergeLo(TimSort.java:781)
	at java.base/java.util.TimSort.mergeAt(TimSort.java:518)
	at java.base/java.util.TimSort.mergeForceCollapse(TimSort.java:461)
	at java.base/java.util.TimSort.sort(TimSort.java:254)
	at java.base/java.util.Arrays.sort(Arrays.java:1516)
	at java.base/java.util.ArrayList.sort(ArrayList.java:1749)
	at java.base/java.util.Collections.sort(Collections.java:177)
	at org.openaudible.convert.SplitJoinUtils.sortByTitle(SplitJoinUtils.java:116)
	at org.openaudible.convert.SplitAudio.isSortedByTitle(SplitAudio.java:375)
	at org.openaudible.convert.SplitAudio.getChapters(SplitAudio.java:333)
	at org.openaudible.convert.SplitAudio.splitByChapter(SplitAudio.java:135)
	at org.openaudible.desktop.swt.manager.SplitJoinGUI.splitOne(SplitJoinGUI.java:245)
	at org.openaudible.desktop.swt.manager.SplitJoinGUI$2.run(SplitJoinGUI.java:168)

Nov. 04, 2021 5:16:10 P.M. org.openaudible.desktop.swt.manager.browser.AudibleBrowser setUrl
INFO: AudibleBrowser setURL: file:/C:/Users/simon/OpenAudible/tmp/split_report.html

**Comment by Butterfly-Dragon** [comment]
i hope this helps

**Comment by openaudible** [comment]
Perfect. Definitely a new bug. If you haven’t already done so, would you mind repeating the bug and send a bug report from the Help menu? Issue is 765 and we can be sure it is fixed. Something unusual about the chapters is unexpected. Thanks!! Will have a fix ASAP.

**Comment by ahokinson** [comment]
Running into something similar for another book. 

```
Nov 04, 2021 6:43:05 PM org.openaudible.convert.SplitAudio splitByChapter
SEVERE: failed to split
java.lang.IllegalArgumentException: Comparison method violates its general contract!
	at java.base/java.util.TimSort.mergeLo(TimSort.java:781)
	at java.base/java.util.TimSort.mergeAt(TimSort.java:518)
	at java.base/java.util.TimSort.mergeCollapse(TimSort.java:448)
	at java.base/java.util.TimSort.sort(TimSort.java:245)
	at java.base/java.util.Arrays.sort(Arrays.java:1516)
	at java.base/java.util.ArrayList.sort(ArrayList.java:1749)
	at java.base/java.util.Collections.sort(Collections.java:177)
	at org.openaudible.convert.SplitJoinUtils.sortByTitle(SplitJoinUtils.java:116)
	at org.openaudible.convert.SplitAudio.isSortedByTitle(SplitAudio.java:375)
	at org.openaudible.convert.SplitAudio.getChapters(SplitAudio.java:333)
	at org.openaudible.convert.SplitAudio.splitByChapter(SplitAudio.java:135)
	at org.openaudible.desktop.swt.manager.SplitJoinGUI.splitOne(SplitJoinGUI.java:245)
	at org.openaudible.desktop.swt.manager.SplitJoinGUI$2.run(SplitJoinGUI.java:168)

**Comment by openaudible** [comment]
I got a good bug report and will have a fix shortly. Thank you.

**Comment by openaudible** [comment]
Appreciate the report. I have a pretty good sense that it is fixed in the latest beta:
https://openaudible.org/beta

**Comment by ahokinson** [comment]
> I have a pretty good sense that it is fixed in the latest beta

I downloaded the beta and can confirm it fixed my issue.

**Comment by Butterfly-Dragon** [comment]
confirmed on my side too.

**Comment by openaudible** [comment]
Fixed in latest release 3.2.1
